### ISSUE TITLE
bug report: shouldn't use tf.nn.sparse_softmax_cross_entropy_with_logits to calculate loss

### ISSUE THREAD

**Issue by Yu-Wu** [AntreasAntoniou/MatchingNetworks#3]
Bug:
Since the predict is based on softmax_similarity, we shouldn't calculate loss with softmax again.

Show:
If using tf.nn.sparse_softmax_cross_entropy_with_logits, the loss will still be high when the accuracy is about 0.99

Solution:
Only one softmax layer exists. Get cross entropy without softmax. The loss will be extramely small when training finished.

**Comment by AntreasAntoniou** [comment]
Thanks for letting me know man. You are correct. I will adjust soon.

**Comment by Yu-Wu** [comment]
For case one shot,  I just removed the softmax layer in calculating similarity. It makes sense since the support set label is a identity matrix.

As for few-shot, I will try the way in https://github.com/tensorflow/tensorflow/issues/2462, although it is not efficient but no choice.

I hope you could find a better solution.
